docs/slashing-protection.md

```diff
@@ -0,0 +1,105 @@
+# Slashing Protection on Finality Provider
+
+In the BTC staking protocol, finality providers run a
+Finality Provider Daemon (`fpd`) to send finality votes to Babylon.
+If a finality provider signs two conflicting blocks (same committed
+randomness but different block hash), all its delegations will be slashed.
+Apart from malicious behavior, honest finality providers face [slashing risks](https://cubist.dev/blog/slashing-risks-you-need-to-think-about-when-restaking)
+due to factors like hardware failures or software bugs.
+Therefore, a proper slashing protection mechanism is required.
+
+Recall that in our system, a finality provider needs to run two daemons:
+1. The finality provider daemon (fpd), which connects to the Babylon node
+   and initiates EOTS signing requests upon a new block to finalize
+2. The EOTS manager daemon (eotsd), which manages the EOTS key and responds to
+   signing requests from the finality provider daemon
+
+The two daemons have different responsibilities to prevent double-signing.
+The protections from the two daemons are complementary to each other.
+
+### Finality provider daemon protection
+
+**Assumption**:
+- The Babylon node the daemon connects to is trusted and responsive.
+- The `finality-provider.db` file is not compromised.
```

Comment:
Instead of noting those as assumptions, should we highlight them with warnings as a key thing to do to ensure the mechanism works?

Reply:
What about **Requirement**?